extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts

```diff
@@ -22,19 +22,77 @@ import { AcctStore } from '../../../../js/common/platform/store/acct-store.js';
 import { PgpHash } from '../../../../js/common/core/crypto/pgp/pgp-hash.js';
+import { UploadedMessageData } from '../../../../js/common/api/account-server.js';
+import { ParsedKeyInfo } from '../../../../js/common/core/crypto/key-store-util.js';
+import { MultipleMessages } from './general-mail-formatter.js';
+import { RecipientType } from '../../../../js/common/api/shared/api.js';
 
 export class EncryptedMsgMailFormatter extends BaseMailFormatter {
 
-  public sendableMsg = async (newMsg: NewMsgData, pubkeys: PubkeyResult[], signingPrv?: Key): Promise<SendableMsg> => {
+  public sendableMsgs = async (newMsg: NewMsgData, pubkeys: PubkeyResult[], signingKey?: ParsedKeyInfo): Promise<MultipleMessages> => {
     if (newMsg.pwd && !this.isDraft) {
       // password-protected message, temporarily uploaded (already encrypted) to:
       //    - flowcrypt.com/api (consumers and customers without on-prem setup), or
       //    - FlowCrypt Enterprise Server (enterprise customers with on-prem setup)
       //    It will be served to recipient through web
-      const { url: msgUrl, externalId } = await this.prepareAndUploadPwdEncryptedMsg(newMsg); // encrypted for pwd only, pubkeys ignored
+      const uploadedMessageData = await this.prepareAndUploadPwdEncryptedMsg(newMsg); // encrypted for pwd only, pubkeys ignored
+      // pwdRecipients that have their personal link
+      const pwdRecipients = Object.keys(uploadedMessageData.emailToExternalIdAndUrl ?? {}).filter(email => !pubkeys.some(p => p.email === email));
       newMsg.pwd = undefined;
-      return await this.sendablePwdMsg(newMsg, pubkeys, { msgUrl, externalId }, signingPrv); // encrypted for pubkeys only, pwd ignored
-    } else if (this.richtext) { // rich text: PGP/MIME - https://tools.ietf.org/html/rfc3156#section-4
+      const collectedAttachments = await this.view.attachmentsModule.attachment.collectEncryptAttachments(pubkeys);
+      const pubkeyRecipients: { [type in RecipientType]?: EmailParts[] } = {};
+      for (const [key, value] of Object.entries(newMsg.recipients)) {
+        if (['to', 'cc', 'bcc'].includes(key)) {
+          const sendingType = key as RecipientType;
+          pubkeyRecipients[sendingType] = value?.filter(emailPart => pubkeys.some(p => p.email === emailPart.email)
+            // pwd recipients that have no personal links go to legacy message
+            || (uploadedMessageData.emailToExternalIdAndUrl || {})[emailPart.email] === undefined);
+        }
+      }
+      const uniquePubkeyRecipientToAndCCs = Value.arr.unique((pubkeyRecipients.to || []).concat(pubkeyRecipients.cc || [])
+        .map(recipient => recipient.email.toLowerCase()));
+      // pubkey recipients should be able to reply to "to" and "cc" pwd recipients
+      const replyTo = (newMsg.recipients.to ?? []).concat(newMsg.recipients.cc ?? [])
+        .filter(recipient => !uniquePubkeyRecipientToAndCCs.includes(recipient.email.toLowerCase()));
```

Comment:
Let's call this `replyToForMessageSentToPubkeyRecipients` :-)

It seems it contains all non-pubkey recipients in the reply-to. Why not all recipients, including the ones with pubkey?

Reply:
Also should be moved inside the `if` just below

Reply:
same for `uniquePubkeyRecipientToAndCCs` since it's only used there

Reply:
> It seems it contains all non-pubkey recipients in the reply-to. Why not all recipients, including the ones with pubkey?

Because the pubkey recipients are seen in `to` and `cc` headers. And we should be careful to not show `bcc`.